shell/platform/linux/fl_gesture_helper.cc

```diff
@@ -0,0 +1,162 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_gesture_helper.h"
+
+struct _FlGestureHelper {
+  GObject parent_instance;
+
+  GList* event_list;
+  GtkWidget* grabbed_widget;
+  GtkWidget* hover_widget;
+  int64_t pointer_id;
+  int32_t pressed_buttons;
+  bool end;
+};
+
+G_DEFINE_TYPE(FlGestureHelper, fl_gesture_helper, G_TYPE_OBJECT)
+
+static void fl_gesture_helper_class_init(FlGestureHelperClass* klass) {}
+
+static void fl_gesture_helper_init(FlGestureHelper* self) {}
+
+FlGestureHelper* fl_gesture_helper_new() {
+  return FL_GESTURE_HELPER(g_object_new(fl_gesture_helper_get_type(), nullptr));
+}
+
+static void free_event(gpointer data) {
+  gdk_event_free(reinterpret_cast<GdkEvent*>(data));
+}
+
+static void send_button_event(FlGestureHelper* self,
+                              GdkEvent* event,
+                              GtkWidget* widget) {
+  if (event->type == GDK_BUTTON_PRESS) {
+    self->end = false;
+    self->pressed_buttons |= 1 << event->button.button;
+  } else if (event->type == GDK_BUTTON_RELEASE) {
+    self->end = true;
+    self->pressed_buttons &= ~(1 << event->button.button);
+  } else {
+    return;
```

Comment:
should we log something here?

Reply:
In fact #42 is not reachable.

Reply:
You could add `g_assert_not_reached()` if you are sure this shouldn't occur.